resolver/master_changer_test.go

```diff
@@ -0,0 +1,91 @@
+package resolver
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	mesos "github.com/mesos/mesos-go/mesosproto"
+)
+
+func assertMastersState(mc *MasterChanger, want []string) error {
```

Comment:
This function is doing two things: a non-blocking channel receive and an assertion. We can factor this better:
```go
// recv receives from a channel in a non-blocking way, returning the received value or nil.
func recv(ch <-chan []string) []string {
    select {
    case val := <-ch:
        return val
    default:
        return nil
    }
}
```

```go
if got := recv(mc.ChangedChan()); !reflect.DeepEqual(got, tt.want) {
    t.Errof("test #%d: got %v, want: %v", i, got, want)
}
```